File: tournaments/isMAC48Address/isMAC48Address.js
function isMAC48Address(inputString) {

    for (var i = 0; i < inputString.length; i++) {
        if (i % 3 === 2) {
            if (inputString[i] !== '-') {
            return false;
            }
        }
        else {
            var sym = inputString[i];
            if (!('0' <= sym && sym <= '9' || 'A' <= sym && sym <= 'F')) {
            return false;
            }
        }
    }
  
    return inputString.length === 17;
}
    

File: tournaments/isMAC48Address/isMAC48Address.cpp
bool isMAC48Address(std::string inputString) {

  for (int i = 0; i < inputString.size(); i++) {
    if (i % 3 == 2) {
      if (inputString[i] != '-') {
        return false;
      }
    }
    else {
      char sym = inputString[i];
      if (!('0' <= sym && sym <= '9' || 'A' <= sym && sym <= 'F')) {
        return false;
      }
    }
  }

  return inputString.size() == 17;
}
